packages/teams-js/src/internal/communication.ts

```diff
@@ -257,6 +259,43 @@ export function sendMessageToParentAsync<T>(
   });
 }
 
+export async function sendMessage<ReceivedFromHost, DeserializedFromHost>(
+  apiVersionTag: string,
+  actionName: string,
+  responseHandler: ResponseHandler<ReceivedFromHost, DeserializedFromHost>,
+  args: ArgsForHost | undefined = undefined,
```

Comment:
I wonder if this could be simplified in a way that doesn't need the custom type (I know, look at me, the custom type advocate saying maybe we don't need a custom type). Ignoring all the other goodness you have here and just focusing on the input arguments, what do you think about something like this?

```
interface ISerializable {
    serialize(): unknown;
}

function isSerializable(arg: unknown): arg is ISerializable {
    return (arg as ISerializable).serialize !== undefined && typeof (arg as ISerializable).serialize === 'function';
}

function sendMessage(actionName: string, args: unknown[]) {
    const outgoingArgs: unknown[] = [];
    for (const arg of args) {
        outgoingArgs.push(isSerializable(arg) ? arg.serialize() : arg);
    }
}
```

Here are a few thoughts I have about it:
- Doesn't require the custom type so theoretically the usage should be simpler
- If a particular action doesn't require any arguments, the caller can just pass [] for the `args` parameter
- Uses what I believe to be more common naming
- Doesn't use `any` so nothing has to be suppressed
- Doesn't need a list of all simple types -- just interrogates each type and says "do you need to be serialized? If so, let me do that. Otherwise, I'll just pass through."

One thing that does seem different but I can't tell if it's good or bad. In `ArgsForHost` you don't list `object` as a simple type, which I think implies that if you want to pass an `object` as an argument, it must implement `ISerializable`. Is that true? If so, is that a necessary restriction? E.g. there are many objects that don't actually need to be serialized because they are just key value pairs of POD (plain old data), e.g.,
```
{
    name: string,
    age: number,
    isStudent: boolean
}
```

To be clear, I'm not saying to get rid of all the other stuff you have (respond handlers, error checker, etc.) -- I haven't looked at those yet, I was just thinking so far on how argument input works.

Reply:
It's an interesting idea. I omitted object from that list very intentionally because I want people who are passing more complicated objects to have to _think_ about whether or not their object requires serialization. I like a lot of the suggestions you have above, but I worry a little bit that if we go that direction fully we'll still be making it pretty easy for people to be passing in things that *should* be serialized without realizing that they *need to be* serialized.

Reply:
I made some changes to move further in this direction while still maintaining the "any objects need to implement ISerializable or you get typescript errors". See what you think!

Reply:
A bit of an aside, do you think making these changes would be better placed in a separate PR instead and then to either keep the `ISerializable` interface as is or to remove it entirely from this PR?

Reply:
I'm not sure what you mean Jay, can you say more? The ISerializable interface is new in this change

Reply:
Aah I was looking over the code and while it is tied together, I think there could be one PR that was just focused on the introduction of the `ISerializable` interface and all the code that touched and another would be the deprecation of the `send` functions and addition of the `callFunctionInHost` ones. The reason I think it would be a good call is that the majority of the feedback on this PR appears to be about the `ISerializable` interface. So for now you can change this block of code to something like the following and then we can follow this PR up with another one that says introducing a new `ISerializable` object.

```javascript
export async function callFunctionInHost(
  apiVersionTag: string,
  functionName: string,
  args: any[],
  errorChecker?: (response: unknown) => response is SdkError,
): Promise<void> {
 //The following probably will need to be changed, but serializing the args is still important
  const argsSafeToTransfer = serializeItemArray(args);
  const [response] = await sendMessageToParentAsync<[SdkError]>(apiVersionTag, functionName, argsSafeToTransfer);

  if ((errorChecker && errorChecker(response)) || (!errorChecker && isSdkError(response))) {
    throw new Error(`${response.errorCode}, message: ${response.message ?? 'None'}`);
  } else if (response !== undefined) {
    // If we receive a response from the host that is not a recognized error type it is an invalid response
    throw new Error(`${ErrorCode.INTERNAL_ERROR}, message: Invalid response from host`);
  }
}
```

Reply:
Talked with Jay some about this. We agreed that in practice it would have made sense to split this up into two PRs: one that just introduced the ISerializable interface and then had certain objects implement it (so reviewers could focus on that item) and then a second one that introduced the new communication functions and made use of them. Unfortunately, because this work started as an FHL project I did not do a great job of planning and sequencing the work into multiple checkins. Because people have already reviewed and read this, splitting it into multiple checkins would not be of much practical use at this point.

Still, good idea and that will teach me to cut corners even during FHL projects :D